File: python/src/aegis/resources/ml.py
"""ML resource — V1 ML inference endpoints."""

from __future__ import annotations

from typing import Any, Dict, List, Optional

from aegis.resources._base import AsyncResource, SyncResource


class SyncML(SyncResource):

    def health(self) -> Dict[str, Any]:
        return self._get("/v1/ml/health")

    def embed(self, text: str, model: Optional[str] = None) -> Dict[str, Any]:
        body: Dict[str, Any] = {"text": text}
        if model is not None:
            body["model"] = model
        return self._post("/v1/ml/embed", json=body)

    def embed_batch(
        self, texts: List[str], model: Optional[str] = None
    ) -> Dict[str, Any]:
        body: Dict[str, Any] = {"texts": texts}
        if model is not None:
            body["model"] = model
        return self._post("/v1/ml/embed/batch", json=body)

    def classify(self, text: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v1/ml/classify", json={"text": text, **kwargs})

    def similarity(
        self, query: str, documents: Optional[List[str]] = None, **kwargs: Any
    ) -> Dict[str, Any]:
        body: Dict[str, Any] = {"query": query}
        if documents is not None:
            body["documents"] = documents
        body.update(kwargs)
        return self._post("/v1/ml/similarity", json=body)


class AsyncML(AsyncResource):

    async def health(self) -> Dict[str, Any]:
        return await self._get("/v1/ml/health")

    async def embed(self, text: str, model: Optional[str] = None) -> Dict[str, Any]:
        body: Dict[str, Any] = {"text": text}
        if model is not None:
            body["model"] = model
        return await self._post("/v1/ml/embed", json=body)

    async def embed_batch(
        self, texts: List[str], model: Optional[str] = None
    ) -> Dict[str, Any]:
        body: Dict[str, Any] = {"texts": texts}
        if model is not None:
            body["model"] = model
        return await self._post("/v1/ml/embed/batch", json=body)

    async def classify(self, text: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v1/ml/classify", json={"text": text, **kwargs})

    async def similarity(
        self, query: str, documents: Optional[List[str]] = None, **kwargs: Any
    ) -> Dict[str, Any]:
        body: Dict[str, Any] = {"query": query}
        if documents is not None:
            body["documents"] = documents
        body.update(kwargs)
        return await self._post("/v1/ml/similarity", json=body)

File: go/ml.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// MLService — V1 ML inference.
type MLService struct {
	t *internal.Transport
}

func (s *MLService) Health(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "GET", "/v1/ml/health", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MLService) Embed(ctx context.Context, text string) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v1/ml/embed", map[string]any{"text": text}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MLService) EmbedBatch(ctx context.Context, texts []string) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v1/ml/embed/batch", map[string]any{"texts": texts}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MLService) Classify(ctx context.Context, text string) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v1/ml/classify", map[string]any{"text": text}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *MLService) Similarity(ctx context.Context, query string, documents []string) (map[string]any, error) {
	body := map[string]any{"query": query}
	if documents != nil {
		body["documents"] = documents
	}
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v1/ml/similarity", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}
